### DISCUSSION TITLE
milvus2.2当load collection的时候，如果内存容量不够全部加载到内存，还能用来搜索不？

### DISCUSSION THREAD

**Discussion by luokuncool** [milvus-io/milvus#21365] [Q&A and General discussion] [answered]
![image](https://user-images.githubusercontent.com/6022635/209286710-d92cdcaa-944e-4b70-b3d3-fe3c144aaefb.png)
加载进度一直停在这里也不动，也不报错。

**Comment by xiaofan-luan** [comment, marked as answer]
> ![image](https://user-images.githubusercontent.com/6022635/209286710-d92cdcaa-944e-4b70-b3d3-fe3c144aaefb.png) 加载进度一直停在这里也不动，也不报错。

加载不成功是不能检索的，可能需要具体看下日志判断原因

**Reply by luokuncool** [reply]
```
milvus-standalone  | [2022/12/23 07:18:14.797 +00:00] [ERROR] [querynode/segment_loader.go:127] ["load failed, OOM if loaded"] [collectionID=438073933266390118] [segmentType=Sealed] ["loadSegmentRequest msgID"=148] [error="load segment failed, OOM if load, collectionID = 438073933266390118, maxSegmentSize = 481 MB, concurrency = 1, usedMemAfterLoad = 10000 MB, totalMem = 10240 MB, thresholdFactor = 0.900000"] [stack="github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:127\ngithub.com/milvus-io/milvus/internal/querynode.(*loadSegmentsTask).Execute\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/load_segment_task.go:81\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:109\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).taskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:131"]
```

看起来就是内存不足了，我看有篇文章说的是2.2开始支持索引存磁盘可以节省内存，不太清楚这个底层逻辑，还是有一些必须加载到内存的东西吗？ 

我有3个字段 定义如下:
```
fields = [
        FieldSchema(name="pk", dtype=DataType.VARCHAR, is_primary=True, auto_id=False, max_length=12), 
        FieldSchema(name="graphic", dtype=DataType.BOOL),
        FieldSchema(name="img", dtype=DataType.FLOAT_VECTOR, dim=512) 
    ]
```
数据量大概有6000万，我应该给服务器准备多大的内存才能load成功呢，不太清楚这个怎么计算。

**Reply by xiaofan-luan** [reply]
> ```
> milvus-standalone  | [2022/12/23 07:18:14.797 +00:00] [ERROR] [querynode/segment_loader.go:127] ["load failed, OOM if loaded"] [collectionID=438073933266390118] [segmentType=Sealed] ["loadSegmentRequest msgID"=148] [error="load segment failed, OOM if load, collectionID = 438073933266390118, maxSegmentSize = 481 MB, concurrency = 1, usedMemAfterLoad = 10000 MB, totalMem = 10240 MB, thresholdFactor = 0.900000"] [stack="github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:127\ngithub.com/milvus-io/milvus/internal/querynode.(*loadSegmentsTask).Execute\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/load_segment_task.go:81\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:109\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).taskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:131"]
> ```
> 
> 看起来就是内存不足了，我看有篇文章说的是2.2开始支持索引存磁盘可以节省内存，不太清楚这个底层逻辑，还是有一些必须加载到内存的东西吗？
> 
> 我有3个字段 定义如下:
> 
> ```
> fields = [
>         FieldSchema(name="pk", dtype=DataType.VARCHAR, is_primary=True, auto_id=False, max_length=12), 
>         FieldSchema(name="graphic", dtype=DataType.BOOL),
>         FieldSchema(name="img", dtype=DataType.FLOAT_VECTOR, dim=512) 
>     ]
> ```
> 
> 数据量大概有6000万，我应该给服务器准备多大的内存才能load成功呢，不太清楚这个怎么计算。

建议至少512G，可能300多G就够了

**Reply by luokuncool** [reply]
好的，多谢！
